### ISSUE TITLE
Request for more drive select / motor control flexibility

### ISSUE THREAD

**Issue by cube1us** [davidgiven/fluxengine#324]
I have run into an issue trying to use my Micropolis drives with my new fluxengine.  I have seen two symptoms:

1) I am more or less able to access one of the drives as 0 (but that might not be drive 1 in micropolis terms)
1) I am unable to access the second drive (see below)
2) If I try to access the second drive as drive 1, that fails, AND the fluxengine board and/or the drive get confused - I have
    to shut everything off and turn it all back on to even get to drive 0

The issue is that I don't have enough control of the motor / drive select lines.

The fluxengine expects this for these pins, from what I can see:

10 MOTEA  12 DIRVEB  14 DRIVEA  16 MOTEB

Whereas the Micropolis drives expect

10 DS1  12 DS2  14 DS3  16 MOTOR

So, as a guess, drive 0 activates pins 10 and 14.  The Micropolis sees pin 10 as DS1 and selects the first drive.  Do you perhaps always drive MOTEA *and* MOTEB?  If so that would explain what that works.  Otherwise, I don't see why it works at all.

But if I then access the second drive, flux engine would presumably activate pins 12 and 16.  I see the drive select light (because fluxengine drive B is DS2), but nothing turns its motor on, so it fails.  That seems odd.

Oddly once I do that, drive 0 no longer works until I reset everything.  It's drive select light doesn't even come on.  It is as though the fluxengine setup can't handle

It would be nice if the software and firmware supported something as an alternative to "drive" (say, driveLines) where one could provide a hex byte to define how these four lines should be managed.  So, for the first micropolis drive it might be 0x9 and for the second drive it would be 0x5.

Of course, I can play with jumpers on the drives and cabling to make this work, but it is a real pain.  I have seen other issues raised that could maybe be dealt with by this generalization, and it would also help with 8" drives.  Also, notably, while my Greaseweazle would not handle the hard sectors, it does not have these particular issues.

JRJ

**Comment by davidgiven** [comment]
I agree. I have here an Atari ST drive, and they're jumpered as drive 0 rather than drive 1 the way PC drives are, and I can't make it work. This all needs an overhaul so that the state of these pins is controlled by the client and not the firmware, to allow arbitrary settings.
